build_tools/github/install.sh

```diff
@@ -1,4 +1,8 @@
 #!/bin/bash -e
 
 python setup.py build
-python setup.py install
+if [ ! -z "$MIN_REQUIREMENTS" ]; then
+    pip install --progress-bar off --upgrade -e .
```

Comment:
@tsalo I'm not sure to follow. I might be wrong, but by looking at the PR it looks like if you do `pip install .` you will install all dependencies in their minimum versions and that you'd have to do `pip install .[dev]` to get the latest versions.
Atm, the minimum dependencies describe the oldest possible configuration that we support (Python 3.6 and minimum deps for all packages).
To me you should be able to pip install with all versions of python above 3.6 without the dev flag (which should only install additional specific packages). Does this make sense?

Reply:
My assumption was that "minimum" dependencies referred to those packages which were necessary to use nilearn, while "dev" was the ones necessary for developers (i.e., additional packages for testing, documentation-building, etc.).

I'm not used to trying to pin different _versions_ of the same packages in different dependency groups. However, I do think it's possible, as long as the minimum group's versions don't conflict with the regular pinned versions.

Reply:
I've added a `min` group with the minimum versions pinned (`==`), while the main dependency group has `>=`. Does that work?

Reply:
Never mind, that doesn't work. Trying to install the min version with `pip install -e .[min]` results in the following:

```
Collecting pandas>=0.24.0
  Using cached pandas-0.24.0.tar.gz (11.8 MB)
ERROR: Cannot install nilearn[min]==0.8.1.dev0 because these package versions have conflicting dependencies.

The conflict is caused by:
    nilearn[min] 0.8.1.dev0 depends on scikit-learn>=0.21.0
    nilearn[min] 0.8.1.dev0 depends on scikit-learn==0.21.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
```

Even though the versions do not actually clash, having two pins like that apparently is not allowed.

Reply:
Actually, it works. The problem was specifically with scikit-learn==0.21.0, as we realized in the previous thread.